src/TideGrantOperation.php

```diff
@@ -0,0 +1,116 @@
+<?php
+
+namespace Drupal\tide_grant;
+
+use Drupal\Component\Utility\Random;
+use Drupal\user\Entity\Role;
+use Drupal\user\Entity\User;
+use Drupal\workflows\Entity\Workflow;
+
+/**
+ * Helper class for install/update ops.
+ */
+class TideGrantOperation {
+
+  /**
+   * Enable Editorial workflow if workflow module is enabled.
+   */
+  public function enableEditorialWorkflow() {
+    $moduleHandler = \Drupal::service('module_handler');
+    if (!\Drupal::service('config.installer')->isSyncing() && $moduleHandler->moduleExists('workflows')) {
+      $editorial_workflow = Workflow::load('editorial');
+      if ($editorial_workflow) {
+        $editorial_workflow->getTypePlugin()
+          ->addEntityTypeAndBundle('node', 'grant');
+        $editorial_workflow->save();
+      }
+    }
+  }
+
+  /**
+   * Add Grant Author user.
+   */
+  public function addGrantAuthorUser() {
+    if (empty(Role::load('grant_author'))) {
+      return;
+    }
+
+    // Create new Grant Author user.
+    $language = \Drupal::languageManager()->getCurrentLanguage()->getId();
+    $user = User::create();
+
+    $random = new Random();
+    $pass = $random->string();
+    $mail = $random->string();
+
+    // Mandatory settings.
+    $user->setPassword($pass);
+    $user->enforceIsNew();
+    $user->setEmail($mail . '@noreply.vic');
+    $user->setUsername('Grant Author');
+    $user->addRole('grant_author');
+
+    // Optional settings.
+    $user->set("init", 'email');
+    $user->set("langcode", $language);
+    $user->set("preferred_langcode", $language);
+    $user->set("preferred_admin_langcode", $language);
+    $user->activate();
+
+    // Save user.
+    $user->save();
+  }
+
+  /**
+   * Assign necessary permissions .
+   */
+  public function assignNecessaryPermissions() {
+    $role_permissions = [
```

Comment:
It would be worth to create a variable to avoid repeating the same permissions. i.e  'create grant content' is added 3 times.

Reply:
didn't really understand this part, sorry. Even if I create a variable, the variable will be used 3 times

Reply:
> didn't really understand this part, sorry. Even if I create a variable, the variable will be used 3 times

If you need to update the value of the variable in future then you only need to update it once instead of 3 times.

Reply:
but the `create grant content` value not going to change as this is associated with the grant module name